test/Microsoft.ML.Tests/Transformers/CustomMappingTests.cs

```diff
@@ -121,5 +122,128 @@ public void TestSchemaPropagation()
 
             Done();
         }
+
+        public class MyStatefulInput
+        {
+            public float Value { get; set; }
+        }
+
+        public class MyState
+        {
+            public HashSet<float> SeenValues;
+        }
+
+        public class MyStatefulOutput
+        {
+            public bool FirstAppearance { get; set; }
+        }
+
+        [CustomMappingFactoryAttribute(nameof(MyStatefulLambda))]
+        public class MyStatefulLambda : StatefulCustomMappingFactory<MyStatefulInput, MyStatefulOutput, MyState>
+        {
+            public override Action<MyStatefulInput, MyStatefulOutput, MyState> GetMapping()
+            {
+                return MyStatefulAction;
+            }
+
+            public override Action<MyState> GetStateInitAction()
+            {
+                return MyStateInit;
+            }
+
+            public static void MyStatefulAction(MyStatefulInput input, MyStatefulOutput output, MyState state)
+            {
+                output.FirstAppearance = !state.SeenValues.Contains(input.Value);
+                state.SeenValues.Add(input.Value);
+            }
+
+            public static void MyStateInit(MyState state)
+            {
+                state.SeenValues = new HashSet<float>();
+            }
+        }
+
+        [Fact]
+        public void TestStatefulCustomMappingTransformer()
+        {
+            string dataPath = GetDataPath("breast-cancer.txt");
+            var source = new MultiFileSource(dataPath);
+            var loader = ML.Data.CreateTextLoader(new[] {
+                new TextLoader.Column("Features", DataKind.Single, 1, 9),
+                new TextLoader.Column("Label", DataKind.String, 0),
+                new TextLoader.Column("Value", DataKind.Single, 2),
+            });
+            var data = loader.Load(source);
+
+            // We create a temporary environment to instantiate the custom transformer. This is to ensure that we don't need the same
+            // environment for saving and loading.
+            var tempoEnv = new MLContext();
+            var customEst = tempoEnv.Transforms.StatefulCustomMapping<MyStatefulInput, MyStatefulOutput, MyState>(MyStatefulLambda.MyStatefulAction, MyStatefulLambda.MyStateInit, nameof(MyStatefulLambda));
+
+            ML.ComponentCatalog.RegisterAssembly(typeof(MyStatefulLambda).Assembly);
```

Comment:
Just a heads up that registering the assembly won't be necessary if this PR gets merged:
https://github.com/dotnet/machinelearning/pull/4989

Since it's modifying the `LambdaTransform` Save and Create methods, it will also be necessary to update `StatefulCustomMappingTransformer` in here to also save the assembly name. My understanding is that there shouldn't be any problem in that PR or in this PR to do this change.

Reply:
Thanks for the heads up. I guess if PR 4989 gets merged before this one, the code of `StatefulCustomMappingTransformer` won't build, so that will remind me to update this :).

---
In reply to: [401497684](https://github.com/dotnet/machinelearning/pull/4676#discussion_r401497684) [](ancestors = 401497684)

Reply:
PR 4989 is merged. 

---
In reply to: [401643916](https://github.com/dotnet/machinelearning/pull/4676#discussion_r401643916) [](ancestors = 401643916,401497684)